oop/oop-kanban/js/script.js

```diff
@@ -0,0 +1,123 @@
+$(function() {
+    
+  function randomString() {
+    var chars = '0123456789abcdefghiklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXTZ';
+    var str = '';
+    for (var i = 0; i < 10; i++) {
+      str += chars[Math.floor(Math.random() * chars.length)];
+    }
+    return str;
+  }
+
+  function Column(name) {
+    var self = this; // useful for nested functions
+
+    this.id = randomString();
+    this.name = name;
+    this.$element = createColumn();
+
+    function createColumn() {
+      var $column = $('<div class="well">').addClass('column');
+      var $columnTitle = $('<h2>').addClass('column-title').text(self.name);
+      var $columnCardList = $('<ul>').addClass('column-card-list');
+      var $columnDelete = $('<button class="btn-danger btn-lg">').addClass('btn-delete').text('x');
+      var $columnAddCard = $('<button class="btn-info btn-lg">').addClass('add-card').text('Add a card');
+
+
+      $columnDelete.click(function() {
+        self.removeColumn();
+      });
+      //Add a note after clicking on the button:
+      $columnAddCard.click(function() {
+        self.addCard(new Card(prompt("Enter the name of the card")));
+      });
+
+      $column.append($columnTitle)
+        .append($columnDelete)
+        .append($columnAddCard)
+        .append($columnCardList);
+      return $column;
+    }
+  }
+
+  Column.prototype = {
+    addCard: function(card) {
+      this.$element.children('ul').append(card.$element);
+    },
+    removeColumn: function() {
+      this.$element.remove();
+    }
+  };
+
+
+  //CARD
+
+  function Card(description) {
+    var self = this;
+
+    this.id = randomString();
+    this.description = description;
+    this.$element = createCard();
+
+
+    function createCard() {
+      var $card = $('<li>').addClass('card');
+      var $cardDescription = $('<p>').addClass('card-description').text(self.description);
+      var $cardDelete = $('<button class="btn-default btn-card">').addClass('btn-delete').text('x');
+
+      $cardDelete.click(function(){
+        self.removeCard();
+      });
+
+      $card.append($cardDelete)
+        .append($cardDescription);
+      return $card;
+    }
+  }
+
+  Card.prototype = {
+    removeCard: function() {
+      this.$element.remove();
+    }
+  }
+
+  var board = {
+    name: 'Kanban Board',
+    addColumn: function(column) {
+      this.$element.append(column.$element);
+      initSortable();
+    },
+    $element: $('#board .column-container')
+  };
+
+  function initSortable() {
+    $('.column-card-list').sortable({
+      connectWith: '.column-card-list',
+      placeholder: 'card-placeholder'
+    }).disableSelection();
+  }
+
+  $('.create-column').click(function(){
+    var name = prompt('Enter a column name');
+    var column = new Column(name);
+    board.addColumn(column);
+
+    // CREATING COLUMNS
+    var todoColumn = new Column('To do');
+    var doingColumn = new Column('Doing');
+    var doneColumn = new Column('Done');
```

Comment:
te kolumny i karty to wstepne wypelnienie kanbana i powinny byc tworzone tylko raz poza klikiem. Teraz za kazdym razem jak tworze kolumne, to tworza mi sie te kolumny z przykladowymi kartami